src/scheduler/scheduler.controller.js

```diff
@@ -0,0 +1,173 @@
+            $scope.totalPages = $scope.jobs.total;
+            LoadingModal.close();
+        });
+
+        JobsService.setListener($scope);
+        JobsService.fetchJobs({});
+
+        $scope.changePageTo = changePageTo;
+
+        function changePageTo(page){
+            JobsService.setParam("page", page);
+            JobsService.fetchJobs();
+        }
+
+        var searchUpdateTimeout;
+        function updateSearch(searchData){
+
+            if(searchUpdateTimeout){
+                clearTimeout(searchUpdateTimeout);
+                searchUpdateTimeout=null;
+            }
+            searchUpdateTimeout = setTimeout(function() {
+                JobsService.setParam("name", searchData.name);
+                JobsService.setParam("activity", searchData.activity);
+                JobsService.setParam("status", searchData.status);
+                JobsService.setParam("timeFrom", searchData.timeFrom);
+                JobsService.setParam("timeTo", searchData.timeTo);
+                JobsService.fetchJobs();
+            }, 1000);
+        }
+
+        $scope.getDetails = function(job) {
+            if ($scope.jobDetails[job.name] !== undefined) {
+            } else {
+                if (!job.uiDefined) {
+                    JobsService.getDetails(job, function(data) {
+                        $scope.jobDetails[job.name] = data;
+                    });
+                }
+            }
+        };
+
+        $scope.updateJob = function(job, updated) {
+            job.activity = updated.activity;
+            job.status = updated.status;
+            job.name = updated.name;
+            job.group = updated.group;
+            job.startDate = updated.startDate;
+            job.nextFireDate = updated.nextFireDate;
+            job.endDate = updated.endDate;
+            job.jobType = updated.jobType;
+            job.info = updated.info;
+            job.uiDefined = updated.uiDefined;
+        };
+
+        $scope.pauseJob = function(job) {
+            ModalFactory.showConfirm("scheduler.confirm.pause", "scheduler.confirm", function(response) {
+                if (response) {
+                    LoadingModal.open();
+                    JobsService.pauseJob(job, function(updated) {
+                        $scope.updateJob(job, updated);
+                        LoadingModal.close();
+                    });
+                }
+            });
+        };
+
+        $scope.resumeJob = function(job) {
+            ModalFactory.showConfirm("scheduler.confirm.resume", "scheduler.confirm", function(response) {
+                if (response) {
+                    LoadingModal.open();
+                    JobsService.resumeJob(job, function(updated) {
+                       $scope.updateJob(job, updated);
+                       LoadingModal.close();
+                    });
+                }
+            });
+        };
+
+        $scope.editJob = function(job) {
+            JobsService.setCurrentJob(job);
```

Comment:
ugh... this pattern of doing this is problematic, and I think I wished it fell off the face of the earth...

Can you add ngDoc to this file, and document this function?

There is a different ticket about making the UI driven by URI state — its really problematic that if you refresh this page you loose context, and it makes it harder to run integration tests because you need to do selenium web driver magic... ideally there would be a parameter in the URL called ID (or a registered pattern)

but lets document for now, and fix it for real later...